apps/designer/app/canvas/shared/design-tokens.ts

```diff
@@ -0,0 +1,20 @@
+import type { DesignToken } from "@webstudio-is/design-tokens";
+import { useEffect, useRef } from "react";
+import { useDesignTokens } from "~/shared/nano-states";
+import { publish } from "~/shared/pubsub";
+
+export const useInitializeDesignTokens = (designTokens: Array<DesignToken>) => {
+  const [, setDesignTokens] = useDesignTokens();
+  const ref = useRef(false);
+  if (ref.current === false) {
+    ref.current = true;
+    setDesignTokens(designTokens);
+  }
+};
```

Comment:
```js
// called once 
useState(() => setDesignTokens(designTokens));
```
As I remember has no issue vs refs has

Reply:
that's a neat trick

Reply:
used it in the places where I used ref before